src/SFML/Window/OSX/AutoreleasePoolWrapper.hpp

```diff
@@ -23,15 +23,54 @@
 //
 ////////////////////////////////////////////////////////////
 
+#ifndef SFML_AUTORELEASEPOOL_HPP
+#define SFML_AUTORELEASEPOOL_HPP
+
 ////////////////////////////////////////////////////////////
-/// \brief Ensure one autorelease pool is available on this thread
-///
+/// Predefine OBJ-C classes
 ////////////////////////////////////////////////////////////
-void ensureThreadHasPool(void);
+#ifdef __OBJC__
+
+@class NSAutoreleasePool;
+typedef NSAutoreleasePool* NSAutoreleasePoolRef;
+
+#else // If C++
+
+typedef void* NSAutoreleasePoolRef;
+
+#endif
 
+namespace sf
+{
 
 ////////////////////////////////////////////////////////////
-/// \brief Drain the thread's pool but keep it alive
+/// \brief Wraps an NSAutoreleasePool that is created when the object is
+///        constructed and is drained when the object is destroyed.
 ///
 ////////////////////////////////////////////////////////////
-void drainThreadPool(void);
+class AutoreleasePool
+{
+public:
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct a new NSAutoreleasePool.
+    ///
+    ////////////////////////////////////////////////////////////
+    AutoreleasePool();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Destructor. Drains the autorelease pool.
```

Comment:
I believe it's fine to just write `Drain the autorelease pool`, that it's a destructor is kind of obvious

Reply:
Agreed. I changed it.